### PR TITLE
Deprecate Sandbox persistence

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala [hunk 1]
```diff
@@ -33,6 +33,12 @@ private[sandbox] object Cli extends SandboxCli {
           s" Also note that instructing $Name to load a scenario will have the side effect of loading _all_ the .dar files provided eagerly (see --eager-package-loading)."
       )
     parser
+      .opt[String]("sql-backend-jdbcurl")
```
Comment (cocreature) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala:36]: @SamirTalwar-DA Is there a good place to share this between the two sadbonxes but not with DAML on SQL?

File: docs/source/tools/sandbox.rst [hunk 2]
```diff
@@ -80,6 +80,9 @@ of the following command line options:
 Running with persistence
 ************************
 
+Note: Running Sandbox with persistence is deprecated. You can use the
```
Comment (bame-da) [inline docs/source/tools/sandbox.rst:83]:
```suggestion
Note: Running Sandbox with persistence is deprecated as of SDK 1.8.0 (16th Dec 2020). You can use the
```

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala [hunk 3]
```diff
@@ -33,6 +33,12 @@ private[sandbox] object Cli extends SandboxCli {
...
+      .optional()
+      .text(
+        s"Deprecated: Use the DAML Driver for PostgreSQL if you need persistence.\nThe JDBC connection URL to a Postgres database containing the username and password as well. If present, $Name will use the database to persist its data.")
```
Comment (bame-da) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala:39]: is it intentionally `$Name`, not `$name` here?

File: ledger/sandbox/src/main/scala/platform/sandboxnext/Cli.scala [hunk 4]
```diff
@@ -30,6 +30,12 @@ private[sandboxnext] object Cli extends SandboxCli {
           + s" You can optionally disable this behavior to bring $Name into line with other ledgers."
       )
     parser
+      .opt[String]("sql-backend-jdbcurl")
+      .optional()
+      .text(
+        s"Deprecated: Use the DAML Driver for PostgreSQL if you need persistence.\nThe JDBC connection URL to a Postgres database containing the username and password as well. If present, $Name will use the database to persist its data.")
```
Comment (bame-da) [inline ledger/sandbox/src/main/scala/platform/sandboxnext/Cli.scala:36]: Is `$Name` correct?

**Review by bame-da — CHANGES_REQUESTED** [review]
Deprecation should also be indicated on https://docs.daml.com/support/component-statuses.html

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala (hunk 3 above)
Comment (cocreature) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/Cli.scala:39]: Yes, `Name` is the argument we pass to the `CommonCli` thingy in which it is then bound to `name`.

File: ledger/sandbox/src/main/scala/platform/sandboxnext/Cli.scala (hunk 4 above)
Comment (cocreature) [reply ledger/sandbox/src/main/scala/platform/sandboxnext/Cli.scala:36]: See above.

File: ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala [hunk 5]
```diff
@@ -36,6 +36,12 @@ private[sql] final class Cli(
         config.copy(jdbcUrl = getEnv(env).orElse(
           throw new IllegalArgumentException(s"The '$env' environment variable is undefined."))))
 
+    opt[String]("sql-backend-jdbcurl")
+      .optional()
+      .text(
+        s"The JDBC connection URL to a Postgres database containing the username and password as well. If present, $name will use the database to persist its data.")
```
Comment (bame-da) [inline ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala:42]: @cocreature I still don't understand why the capitalization of `$name` is different here, but if you say it's all correct, I believe you.

**Review by bame-da — APPROVED** [review]
Thank you!

File: ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala (hunk 5 above)
Comment (cocreature) [reply ledger/daml-on-sql/src/main/scala/on/sql/Cli.scala]: Ah that one is wrong ind fails to build, fixed. Basically `name` in `CommonCli`, `Name` outside.

**Review by stefanobaghino-da — APPROVED** [review]
🎉

**Pull request merged**